hooks/useStore.js

```diff
@@ -43,6 +43,19 @@ const useStore = create(
             };
           });
         },
+
+        editRecipe: currentRecipe => {
+          set(state => {
+            return {
+              recipeList: state.recipeList.map(recipe =>
+                recipe.id === currentRecipe.id
+                  ? { ...recipe, ...currentRecipe }
+                  : recipe
```

Comment:
Das hier ist nicht falsch und in deinem scope kannst du das auch so lassen.  Ich will hier nur einmal die Alternative aufzeigen und argumentieren, warum ich das anders machen würde.
zu erst meine Version:
```
recipeList: {
const index = state.recipeList.findIndex(recipe => recipe.id === currentRecipe.id)
recipeList[index] = currentRecipe;
}
```
Das ganze hat folgenden Vorteil. Das Map muss jeden einzelnen eintrag verarbeiten und ihn prüfen, selbst wenn er schon den zutreffenden eintrag gefunden hat, macht er das trotzdem noch weiter für alle anderen einträge. FindIndex, hört auf sobald er einen Index gefunden hat und somit ist dieser vom rechenaufwand weniger teuer und schneller

Reply:
das merke ich mir und lass es daher noch offen